EIPS/eip-3777.md

```diff
@@ -0,0 +1,312 @@
+### UpdateHashedProof
+### Query
+## Reference Implementation
+### TimeMarker
+### Dynamic NFT
+A dynamic emoji story NFT: 
+ * 📜The story is randomized emoji dynamic generated and stored on-chain.
+ * Script, text, plot, and other functionality are intentionally omitted for others to interpret.
+ * 🆕Each NFT updates the story automatically every 42 hours (default) over time. However, its rarity are not affected.
+ * Feel free to use emoji stories hints sheet in any way you want.
+ * Website: https://www.storyproject.xyz
+ * Contract: https://etherscan.io/address/0x1510b4bc4f6d19b17b00f2d330cdc4b2bc1f470a#code
+
+### Governance voting
+## Security Considerations
+1. Inherit the security based on the ERC721 standard.
+    1. Related security risks have been resolved by ERC721 and have been operating stably for many years without security issues.
+    2. There may be some security risks: At present, ERC721 will execute the _checkOnERC721Received function when calling safeTransfer and _mint to call the onERC721Received function of the external contract, which brings the risk of external callback reentry. We can implement specific functions according to the specific function. The execution content adds a modifier to prevent reentry.
+2. security for addition tpNFT functional 
+    1. Set tpNFT to immutable state (irreversible): The owner of tpNFT can set it to be immutable. After that, the series of update methods of this tpNFT will not be able to call, so that even if the NFT is lost for various reasons, it will not be modified.
+    2. When updating the data of tpNFT, it is prohibited to directly modify the TimeProof struct in the _timeProofs mapping, must use the three internal methods _updateURI, _updateDescription, and _updateHashProof provided by the standard to update the data of tpNFT so that the updated content and the updated timestamp are in the same atom transaction and executed correctly.
+3. Keccak256 is selected as the function of Hash, it is widely used and its safety has been tested for a long time.
+    1. Anti-collision: Can not find two different data with the same Hash value.
+    2. Irreversible: Raw raw data cannot be obtained by reverse calculation of Hash value.
+    3. Leakage risk: We provide a simple function [genHashProof] to pass in the string and generate the hash value. It should be clarified that this function is for reference only. The preimage may be leaked due to RPC and other reasons when using this method of deploying the contract on the chain. Therefore for safety, if your preimage is very important, it is recommended that you copy the source code to the local offline or use other SDK to generates hash values ​​in a local offline environment.
+## References
+
+**Standards**
+
+1. [ERC-20](./eip-20.md) Token Standard.
+2. [ERC-165](./eip-165.md) Standard Interface Detection.
+3. [ERC-721](./eip-721.md) Non-Fungible Token Standard.
+## Copyright
+Copyright and related rights waived via CC0.

```

Comment:
```suggestion
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```

Reply:
Thanks for guiding the revision, it has been corrected, please check again.